bigbluebutton-html5/imports/ui/components/presentation/cursor/component.jsx

```diff
@@ -1,81 +1,289 @@
+    return ((attribute * widthRatio) / 100) / physicalWidthRatio;
+  }
 
-  // x Position of the view box
-  viewBoxX: PropTypes.number.isRequired,
+  static invertScale(attribute, propsObj) {
+    const { widthRatio, physicalWidthRatio } = propsObj;
+    return ((attribute * physicalWidthRatio) * 100) / widthRatio;
+  }
 
-  // y Position of the view box
-  viewBoxY: PropTypes.number.isRequired,
+  static getCursorCoordinates(props) {
+    // main cursor x and y coordinates
+    const x = (props.cursorX / 100) * props.slideWidth;
+    const y = (props.cursorY / 100) * props.slideHeight;
 
-  // Slide to view box width ratio
-  widthRatio: PropTypes.number.isRequired,
+    return {
+      x,
+      y,
+    };
+  }
 
-  // Defines the cursor x position
-  cursorX: PropTypes.number.isRequired,
+  static getFillAndLabel(props) {
+    const { presenter, isMultiUser } = props;
 
-  // Defines the cursor y position
-  cursorY: PropTypes.number.isRequired,
+    const obj = {
+      fill: 'green',
+      displayLabel: false,
+    };
 
-  /**
-   * Defines the cursor fill colour
-   * @defaultValue 'red'
-   */
-  fill: PropTypes.string,
+    if (presenter) {
+      obj.fill = 'red';
+    }
 
-  /**
-   * Defines the cursor radius
-   * @defaultValue 5
-   */
-  radius: PropTypes.number,
-};
+    if (isMultiUser) {
+      obj.displayLabel = true;
+    }
 
-const defaultProps = {
-  fill: 'red',
-  radius: 5,
-};
+    return obj;
+  }
 
-export default class Cursor extends Component {
-  constructor(props) {
-    super(props);
+  static getScaledSizes(props) {
+    const scaleFactor = props.widthRatio / 100 / props.physicalWidthRatio;
+
+    return {
+      // Adjust the radius of the cursor according to zoom
+      // and divide it by the physicalWidth ratio, so that svg scaling wasn't applied to the cursor
+      finalRadius: props.radius * scaleFactor,
+      // scaling the properties for cursorLabel and the border (rect) around it
+      cursorLabelText: {
+        textDY: props.cursorLabelText.textDY * scaleFactor,
+        textDX: props.cursorLabelText.textDX * scaleFactor,
+        fontSize: props.cursorLabelText.fontSize * scaleFactor,
+      },
+      cursorLabelBox: {
+        xOffset: props.cursorLabelBox.xOffset * scaleFactor,
+        yOffset: props.cursorLabelBox.yOffset * scaleFactor,
+        // making width and height a little bit larger than the size of the text
+        // received from BBox, so that the text didn't touch the border
+        width: (props.labelBoxWidth + 3) * scaleFactor,
+        height: (props.labelBoxHeight + 3) * scaleFactor,
+        strokeWidth: props.cursorLabelBox.labelBoxStrokeWidth * scaleFactor,
+      },
+    };
   }
 
-  render() {
+  componentWillMount() {
+    const cursorCoordinate = Cursor.getCursorCoordinates(this.props);
+    const { fill, displayLabel } = Cursor.getFillAndLabel(this.props);
+    const _scaledSizes = Cursor.getScaledSizes(this.props);
```

Comment:
this.scaledSizes = Cursor.getScaledSizes(this.props);
this.cursorCoordinate = Cursor.getCursorCoordinates(this.props);;